### DISCUSSION TITLE
Why use auto-typing and not simply pasting in?

### DISCUSSION THREAD

**Discussion by KodeXall** [keepassxreboot/keepassxc#9259] [Q&A]
I have two questions, I think somehow related:

**1]** If some keylogger records every stroke of keyboard a user makes - then would not coping the whole password from _KeePassXC_ and just executing `CTRL+V `be a safer method of delivering the password?

Or do keyloggers also record what is in the _Clipboard_ of _Windows_ thus it would not increase security?

Or is it the job of _KeePassXC-Browser_ add-on to secure communication? But then again- not every one uses _Firefox_ and some people store in _KeePassXC_ passwords for also other databases or e.g. for securing access to their antivirus settings

I sometimes make a stupid mistake and send my user and password to the wrong window or switch from a proper window to some other as I do not see if my very long password already ended entering itself in its entirety - and as a result I get a catastrophic cascade of commands executed in e.g. _Thunderbird_ on my e-mails. If all what was done was a simple `CTRL+V` then no fatal result would occur

**[DISREGARD THE FOLLOWING:]**

**2]** Why does setting

**_Tools > Setting > General > Auto-Type typing delay > 0 ms_**

do not make the user and password be immediately pasted in? It makes it being written very fast but nevertheless not speed-light fast

Why make a user wait? To give user a chance to stop the password being entered to a wrong site / window / program? If so, then this delay such be significant, something like a 333ms - which would make a password consisting of 100 sign take 30 seconds to complete, which would be ridiculous

**Comment by KodeXall** [comment]
> **[DISREGARD THE FOLLOWING:]**
>
> **2]** Why does setting
>
> **_Tools > Setting > General > Auto-Type typing delay > 0 ms_**

Yep

I would only like to get an answer to the first question, as this second issue somehow went away on its own

**Reply by droidmonkey** [reply]
There is no such thing as a key logger. That may have been true in the 90s, not now. Today's malware literally records and steals every bit of information on your computer. This includes taking screenshots of everything you do, your clipboard, what you type, everything.  If you were to somehow get this installed on your computer it is game over. KeePassXC is not going to be able to do anything to prevent data exfiltration. Your best defense is an antivirus/Anti-Malware capability and DO NOT download anything you don't trust.

**Reply by KodeXall** [reply]
Thank you for enlightening me
